lib/common/acl.c

```diff
@@ -781,3 +790,363 @@ pcmk__update_acl_user(xmlNode *request, const char *field,
 
     return requested_user;
 }
+
+#define ACL_NS_PREFIX "http://clusterlabs.org/ns/pacemaker/access/"
+#define ACL_NS_Q_PREFIX  "pcmk-access-"
+#define ACL_NS_Q_WRITABLE (const xmlChar *) ACL_NS_Q_PREFIX   "writable"
+#define ACL_NS_Q_READABLE (const xmlChar *) ACL_NS_Q_PREFIX   "readable"
+#define ACL_NS_Q_DENIED   (const xmlChar *) ACL_NS_Q_PREFIX   "denied"
+
+static const xmlChar *NS_WRITABLE = (xmlChar *) ACL_NS_PREFIX "writable";
+static const xmlChar *NS_READABLE = (xmlChar *) ACL_NS_PREFIX "readable";
+static const xmlChar *NS_DENIED =   (xmlChar *) ACL_NS_PREFIX "denied";
+
+static int
+pcmk__eval_acl_as_namespaces_2(xmlNode *xml_modify)
+{
+
+    static xmlNs *ns_recycle_writable = NULL,
+                 *ns_recycle_readable = NULL,
+                 *ns_recycle_denied = NULL;
+    static const xmlDoc *prev_doc = NULL;
+
+    xmlNode *i_node = NULL;
+    const xmlChar *ns;
+    int ret = 0;
+
+    if (prev_doc == NULL || prev_doc != xml_modify->doc) {
+        prev_doc = xml_modify->doc;
+        ns_recycle_writable = ns_recycle_readable = ns_recycle_denied = NULL;
+    }
+
+    for (i_node = xml_modify; i_node != NULL; i_node = i_node->next) {
+        switch (i_node->type) {
+        case XML_ELEMENT_NODE:
+            pcmk__set_xml_flag(i_node, xpf_tracking);
```

Comment:
This function was renamed to pcmk__set_xml_doc_flag(), and you renamed the xpf_* values